pkg/doc/verifiable/presentation.go

```diff
@@ -327,6 +340,8 @@ func decodeRawPresentation(vpData []byte, vpOpts *presentationOpts) ([]byte, *ra
 			return nil, nil, fmt.Errorf("decoding of Verifiable Presentation from JWS: %w", err)
 		}
 
+		rawCred.proved = true
```

Comment:
not sure `proved` and its logic adds any benefit since its set true only if decode...() didn't return an error.

the assumption is rawCred will be the presentation proof, is this right? so if this function returns no error, then rawCred is the proof. 

Please clarify the need of this new bool.

Reply:
At `decodeRawPresentation()`, `proved` is set to `true` in case of **JWT**. If it's just JSON, `proved=false` and we need to check for embedded prove presence.
It's an interesting question,  can an embedded credential be proof of the presentation. It should be answered at #862 .